programs/svm-spoke/src/lib.rs

```diff
@@ -194,11 +194,18 @@ pub mod svm_spoke {
             proof,
         )
     }
-    pub fn bridge_tokens_to_hub_pool(
-        ctx: Context<BridgeTokensToHubPool>,
+    pub fn bridge_tokens_to_hub_pool<'info>(
+        ctx: Context<'_, '_, '_, 'info, BridgeTokensToHubPool<'info>>,
         amount: u64,
     ) -> Result<()> {
-        ctx.accounts.bridge_tokens_to_hub_pool(amount, &ctx.bumps)?;
+        let context = Context {
```

Comment:
Workaround to be able to pass the context required by `emit_cpi!`

Reply:
I think it would be more consistent with other instructions to use free functions and pass ctx as parameter.

Reply:
that should also hopefully avoid account cloning below